runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -243,26 +250,32 @@ class MM_ObjectAllocationAPI
 				}
 
 				/* Initialize the object */
-				objectHeader->clazz = (j9objectclass_t)(UDATA)arrayClass;
-				objectHeader->size = size;
-
-				if (initializeSlots) {
-					memset(objectHeader + 1, 0, dataSize);
+				if (J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(currentThread)) {
+					J9IndexableObjectContiguousCompressed *header = (J9IndexableObjectContiguousCompressed*)objectHeader;
+					header->clazz = (j9objectclass_t)(UDATA)arrayClass;
```

Comment:
Is this cast correct actually? Should it be `(U_32)` or something 32-bit class specific?

Reply:
Fixed.